components/delegations/columns.tsx

```diff
@@ -0,0 +1,265 @@
+'use client';
+
+import { type Row, type RowData, createColumnHelper, filterFns, sortingFns } from '@tanstack/react-table';
+import type { Delegation } from 'lib/delegate/DelegatePlatform';
+import { useTranslations } from 'next-intl';
+import { useMemo } from 'react';
+import ContractCell from './cells/ContractCell';
+import ControlsCell from './cells/ControlsCell';
+import DelegateCell from './cells/DelegateCell';
+import DelegationTypeCell from './cells/DelegationTypeCell';
+import DelegatorCell from './cells/DelegatorCell';
+import PlatformCell from './cells/PlatformCell';
+
+export enum ColumnId {
+  TYPE = 'Delegation Type',
+  DELEGATOR = 'Delegator',
+  DELEGATE = 'Delegate',
+  CONTRACT = 'Contract',
+  PLATFORM = 'Platform',
+  CONTROLS = 'Controls',
+}
+
+// Accessors for delegation data - these can be used for sorting and filtering
+export const accessors = {
+  type: (delegation: Delegation) => delegation.type,
+  delegator: (delegation: Delegation) => delegation.delegator,
+  delegate: (delegation: Delegation) => delegation.delegate,
+  contract: (delegation: Delegation) => delegation.contract,
+  platform: (delegation: Delegation) => delegation.platform,
+};
+
```

Comment:
Because these are "simple" accessors, we can also just use the string (e.g. `'type'`) below. Not super important to change though.